### PR TITLE
Massive Update - Remove old storage check, add new storage and resource check to all scripts - Remove downscaling with pct set

### PR DESCRIPTION
**Pull request by MickLesk**
## Description
Massive Update (Remove StorageCheck (old) + Add Ressource & StorageCheck, Remove Whiptail, Remove PCT Ressource Set

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- (https://github.com/community-scripts/ProxmoxVE/pull/249)

### CODE REVIEW

**Comment by remz1337** [comment]
Awesome work. I'm supportive of the usage of the new centralized functions (`check_ressources`). However, can we discuss the  PCT set resources? Personnaly, I hate having my LXC overprovisioned. I would prefer to keep that part.

**Comment by MickLesk** [comment]
Yes we can discuss, but the right thing to do in my opinion would be to make a small fix PR, with proper resources. In the end, Proxmox and the LXC only use what it actually needs :-) if it doesn't need it, it doesn't use it. (Unlike with Homeassistant VM 😂)

File: ct/homeassistant-core.sh [hunk 1]
```diff
@@ -65,6 +65,8 @@ function update_script() {
     "3" "Install FileBrowser" OFF \
     3>&1 1>&2 2>&3)
   header_info
+  check_container_storage
+  check_container_resources
```
Comment (havardthom) [inline ct/homeassistant-core.sh:69]: should these before installation check like other scripts?

File: ct/homeassistant.sh [hunk 2]
```diff
@@ -64,6 +64,8 @@ function update_script() {
     "4" "Install FileBrowser" OFF \
     3>&1 1>&2 2>&3)
   header_info
+  check_container_storage
+  check_container_resources
```
Comment (havardthom) [inline ct/homeassistant.sh:68]: should these before installation check like other scripts?

File: ct/linkwarden.sh [hunk 3]
```diff
@@ -65,7 +65,7 @@ if [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]] || [[ ! -f /opt/${APP}
 
   msg_info "Updating ${APP} to ${RELEASE}"
   cd /opt
-  mv /opt/linkwarden /opt/linkwarden_bak
+  mv /opt/linkwarden/.env /opt/.env
```
Comment (havardthom) [inline ct/linkwarden.sh:68]: This change should have been done in a separate PR

File: ct/node-red.sh [hunk 4]
```diff
@@ -59,6 +59,8 @@ UPD=$(whiptail --backtitle "Proxmox VE Helper Scripts" --title "SUPPORT" --radio
   "2" "Install Themes" OFF \
   3>&1 1>&2 2>&3)
 header_info
+check_container_storage
+check_container_resources
```
Comment (havardthom) [inline ct/node-red.sh:63]: should these before installation check like other scripts?

File: ct/plex.sh [hunk 5]
```diff
@@ -60,6 +60,8 @@ UPD=$(whiptail --backtitle "Proxmox VE Helper Scripts" --title "SUPPORT" --radio
   3>&1 1>&2 2>&3)
 
 header_info
+check_container_storage
+check_container_resources
```
Comment (havardthom) [inline ct/plex.sh:64]: Move before app installation check

File: ct/podman-homeassistant.sh [hunk 6]
```diff
@@ -65,6 +65,8 @@ function update_script() {
   "4" "Remove ALL Unused Images" OFF \
   3>&1 1>&2 2>&3)
 header_info
+check_container_storage
+check_container_resources
```
Comment (havardthom) [inline ct/podman-homeassistant.sh:69]: Move before app installation check

**Review by havardthom — DISMISSED** [review]
Nice, only comments on moving the functions before app installation and whiptail in some scripts. I tested moving `header_info` before whiptail in cockpit script locally and it worked fine.
Also have you tested that `check_container_storage` and `check_container_resources` works on all supported distributions? 
```
debian 11
debian 12
ubuntu 20.04
ubuntu 22.04
ubuntu 24.04
```

**Comment by MickLesk** [comment]
Jep, local it works for Fedora and co too. Thats a later PR. For now only all Debian & Ubuntu Systems

**Comment by MickLesk** [comment]
@havardthom all done.

File: ct/cronicle.sh [hunk 7]
```diff
@@ -53,13 +53,15 @@ function default_settings() {
 }
 
 function update_script() {
+header_info
+check_container_storage
+check_container_resources
 UPD=$(whiptail --backtitle "Proxmox VE Helper Scripts" --title "SUPPORT" --radiolist --cancel-button Exit-Script "Spacebar = Select" 11 58 2 \
-  "1" "Update ${APP}" ON \
+  "1" "Upgrade ${APP}" ON \
```
Comment (havardthom) [inline ct/cronicle.sh:60]:
```suggestion
  "1" "Update ${APP}" ON \
```
Why change, we call it update everywhere else

**Review by havardthom — DISMISSED** [review]

**Review by Mellowlynx — APPROVED** [review]

**Comment by newzealandpaul** [comment]
> Awesome work. I'm supportive of the usage of the new centralized functions (`check_ressources`). However, can we discuss the PCT set resources? Personnaly, I hate having my LXC overprovisioned. I would prefer to keep that part.

...

> Yes we can discuss, but the right thing to do in my opinion would be to make a small fix PR, with proper resources. In the end, Proxmox and the LXC only use what it actually needs :-) if it doesn't need it, it doesn't use it. (Unlike with Homeassistant VM 😂)

For setups with limited resources, it does make sense to not over-provision. If for example I try a script that ends up consuming a lot more storage when in use (this happened to me a few days ago) I would rather that new LXC crashed, than my long running LXCs.

**Comment by newzealandpaul** [comment]
Code looks fine and will approve, but I don't think our policy should be over provisioning LXC.

**Review by newzealandpaul — APPROVED** [review]
